src/components/DepositDialog.jsx

```diff
@@ -31,32 +64,37 @@ export default function DepositDialog({ onClose, baseOrQuote }) {
   if (!coinMint) {
     return null;
   }
+
   return (
     <Modal
       title={depositCoin}
       visible={!!coinMint}
       onOk={onClose}
       onCancel={onClose}
     >
-      <div style={{ paddingTop: '20px' }}>
-        <p style={{ color: 'white' }}>Mint address:</p>
-        <p style={{ color: 'rgba(255,255,255,0.5)' }}>{coinMint.toBase58()}</p>
-        <div>
-          <p style={{ color: 'white' }}>Deposit address:</p>
-          <p style={{ color: 'rgba(255,255,255,0.5)' }}>
-            {account ? (
-              account.pubkey.toBase58()
-            ) : (
-              <>
-                Visit{' '}
-                <Link external to={providerUrl}>
-                  {providerName}
-                </Link>{' '}
-                to create an account for this mint
-              </>
-            )}
-          </p>
-        </div>
+      <div>
+        <p style={{ color: 'white' }}>Deposit address:</p>
+        <p style={{ color: 'rgba(255,255,255,0.5)' }}>
+          {account ? (
+            account.pubkey.toBase58()
+          ) : (
+            <>
+              {connected ? (
+                <ActionButton
+                  block
+                  size="large"
+                  loading={isCreatingTokenAccount}
+                  onClick={doCreateTokenAccount}
+                >
+                  Create token account
+                </ActionButton>
+              ) : (
+                <WalletConnect />
+              )}
+            </>
+          )}
+        </p>
+        {account && <QRCode value={account.pubkey.toBase58()} />}
```

Comment:
QR codes need a white margin around them to work; passing in `includeMargin` to `QRCode` should work